boudicca.base/common-model/src/main/kotlin/base/boudicca/model/structured/KeyFilter.kt

```diff
@@ -0,0 +1,26 @@
+package base.boudicca.model.structured
+
+/**
+ * represents a parsed KeyFilter of a Key-Value pair which consists of the name and all the variants (which are sorted canonically)
+ * similar to a Key, but a KeyFilter also allows "*" as a propertyName and "*", "" (the empty string) as values for variants
+ */
+class KeyFilter(name: String, variants: List<Variant> = emptyList()) : AbstractKey<KeyFilter>(name, variants) {
+
+    companion object {
+        fun parse(keyFilter: String): KeyFilter {
+            return KeyUtils.parseKeyFilter(keyFilter)
+        }
+
+        fun builder(propertyName: String): KeyFilterBuilder {
+            return KeyFilterBuilder(propertyName)
+        }
+    }
+
+    override fun toBuilder() = KeyFilterBuilder(name).withVariants(variants)
+
+    class KeyFilterBuilder(name: String) : AbstractKeyBuilder<KeyFilter>(name) {
```

Comment:
So if we did that the Kotlin way we would get rid of the "toBuilder" function and move the KeyFilterBuilder to a separate file.

Then define a `fun keyFilter(keyFilter: KeyFilter, init KeyFilterBuilder.() -> {})` (or call it modifyKeyFilter if you want and keyFilter for creating new filters)